akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -65,9 +67,10 @@ private[io] final class AsyncDnsResolver(
           resolveWithResolvers(name, mode, resolvers)
             .map { resolved =>
               if (resolved.records.nonEmpty) {
-                val minTtl = resolved.records.minBy[Duration](_.ttl.value).ttl
+                val minTtl = (positiveCachePolicy +: resolved.records.map(_.ttl)).min
                 cache.put((name, mode), resolved, minTtl)
-              }
+              } else if (negativeCachePolicy != Never) cache.put((name, mode), resolved, negativeCachePolicy)
+              log.debug(s"{} resolved {}", mode, resolved)
```

Comment:
I'd like some test coverage of both these changes as well, and not only the setting. Could be done by injecting a `TestProbe` actor ref instead of an actual resolver creating the AsyncDnsResolver actor in the test and verify interactions or no interactions depending on the setting.